### ISSUE TITLE
Change minter-statefile location

### ISSUE THREAD

**Issue by grosscol** [mlibrary/umrdr#10]
The default /tmp minter statefile location is a poor choice since it will be blown away on reboot which will lead to reused IDs.

A better choice for production would be `/etc/umrdr` or `/etc/deepbluedata`

**Comment by grosscol** [comment]
For dev and testing /tmp should be a sufficient location.  For production use `/var/deepbluedata/minter-statefile`.  Creating the directory and modifying the permissions to allow the application access will have to be part of the deployment or pre-deployment actions.

**Comment by grosscol** [comment]
closed by #38

**Comment by jgmorse** [comment]
Should this be in a non-app-specific location, since other Hydra apps will presumably use the same ID minter?

**Comment by grosscol** [comment]
tl;dr each application needs it's own minter statefile.

No two apps should be using the same minter statefile.  The file maintains the state of the logic for creating new ids.  The issue is the default location of /tmp gets blown away, and the state that reverts to the initial state.  There is not sufficient entropy in the way ids are minted to avoid collisions.  So for an application to continue to mint ids for it's fedora objects, the minter statefile needs to be persistent.

Two instances of an applications using the same repository would both have to access the same statefile to avoid conflicts.  If we end up doing multi-site replication, there are strategies that can mitigate having to wait to get a lock on the minter-file.

Also, I don't expect other hydra applications to use the umrdr code.  If there is anything useful, like devise auth with cosign, we should extract that to it's own gem.

**Comment by aelkiss** [comment]
Some additional info that might allay @jgmorse's concerns here: the identifier scheme Sufia uses to mint identifiers (NOID) supports minter-specific prefixes that we can use to avoid collisions between minters maintaining separate statefiles. We will need a way to manage those prefixes to ensure they aren't re-used. Continuing discussion around identifiers will happen at #9 .
